source/routing/get_stop_times.cpp

```diff
@@ -60,9 +60,9 @@ std::vector<datetime_stop_time> get_stop_times(const routing::StopEvent stop_eve
         auto st = next_st.next_stop_time(stop_event, jpp_idx,
                                          dt, clockwise, rt_level,
                                          accessibilite_params.vehicle_properties, true);
-
         if (st.first) {
-            next_requested_dt.push({jpp_idx, st.first, st.second});
+            auto diff = clockwise ? st.first->get_boarding_duration() : -1 * st.first->get_alighting_duration();
+            next_requested_dt.push({jpp_idx, st.first, st.second + diff});
```

Comment:
Needs a test too I guess :). On departure_boards there is a sort after calling get_stop_times but it doesn't seems to be the case everywhere..